src/gencost/waterfall.py

```diff
@@ -1174,105 +1230,148 @@ def get_860_by_x(
-            "carbon_capture": "capacity_mw",
-            "fluidized_bed_tech": "capacity_mw",
-            "pulverized_coal_tech": "capacity_mw",
-            "stoker_tech": "capacity_mw",
-            "other_combustion_tech": "capacity_mw",
-            "subcritical_tech": "capacity_mw",
-            "supercritical_tech": "capacity_mw",
-            "ultrasupercritical_tech": "capacity_mw",
-            "age_from_report_year": "capacity_mw",
-            "avg_age_from_report_year": "capacity_mw",
-            "current_avg_age": "capacity_mw",
-            "age_of_observation": "capacity_mw",
-            "age_relative_to_avg": "capacity_mw",
-            "pollution_control_costs_per_kw": "capacity_mw",
-        }
-
-        if age_year is not None:
-            age_year_str = dt.strptime(f"12-1-{age_year}", "%m-%d-%Y")
-        else:
-            age_year_str = dt.utcnow()
-        return (
-            merged.query("_merge == 'both'")
             .assign(
                 age_from_report_year=lambda x: (
                     x["report_date"] - x["generator_operating_date"]
                 ).dt.days
                 / 365.25,
-                avg_age_from_report_year=lambda x: x.groupby(
-                    ["plant_id_eia", subplant_id_col]
-                )["age_from_report_year"].transform("mean"),
+                avg_age_from_report_year=lambda x: x.age_from_report_year,
                 current_age=lambda x: (
                     age_year_str - x["generator_operating_date"]
                 ).dt.days
                 / 365.25,
-                current_avg_age=lambda x: x.groupby(["plant_id_eia", subplant_id_col])[
-                    "current_age"
-                ].transform("mean"),
+                current_avg_age=lambda x: x.current_age,
                 age_of_observation=lambda x: (age_year_str - x["report_date"]).dt.days
                 / 365.25,
                 age_relative_to_avg=lambda x: x["current_age"]
                 - x["avg_age_from_report_year"],
             )
-            .astype({k: float for k in wtavg_dict})
-            .fillna({k: 0.0 for k in wtavg_dict})
-            .drop(columns=["_merge"])
-            .pipe(
-                sum_and_weighted_average_agg,
-                by=[
-                    "plant_id_eia",
-                    subplant_id_col,
-                    pd.Grouper(key="report_date", freq="YS"),
-                ],
-                sum_cols=["capacity_mw"],
-                wtavg_dict=wtavg_dict,
-            )
-            .astype({"plant_id_eia": "Int64", subplant_id_col: "Int64"})
         )
 
+        if subplant_id_col == "generator_id":
+            # have to cast True, False, and NA from tech cols in 860 as 0 and 1
+            # since at the PF level, this is done in wt avg
+
+            merged = self.tech_cols_dummy(merged)
+
+            if merge_only:
+                return merged
+
+            return merged.query('_merge == "both"')[GET_860_GEN_COLS].pipe(add_ba_code)
```

Comment:
I don't think the merge_only return is needed because we're not checking merge stats here. Also because there is no indicator column created above, the query throws an error.

Instead of returning `merged` when if `merge_only`, might want to log a warning that `merge_only` doesn't do anything when getting results by generator.